src/base/bittorrent/sessionimpl.cpp

```diff
@@ -5565,6 +5625,44 @@ void SessionImpl::handleTorrentDeleteFailedAlert(const lt::torrent_delete_failed
     m_removingTorrents.erase(removingTorrentDataIter);
 }
 
+void SessionImpl::handleTorrentNeedCertAlert(const lt::torrent_need_cert_alert *p)
+{
+    const QString torrentID = (TorrentID {p->handle.info_hash()}).toString();
+
+    TorrentImpl *const torrent = m_torrents.value(p->handle.info_hash());
+    if (!torrent)
+        return;
+
+    const Path baseDir = sslPeersCertificatesDirectory();
+    if (baseDir.exists())
+    {
+
+        Path cert = baseDir / Path {torrentID};
```

Comment:
```suggestion
        Path cert = baseDir / Path(torrentID);
```

Reply:
etc. where it is `Class(constructor, params)` and there is no ambiguity during compilation.

Reply:
Also I would use more explicit variable names, e.g. `certPath` since it represents a path, not a certificate.